modules/core/04-channel/types/upgrade_test.go

```diff
@@ -0,0 +1,16 @@
+package types_test
+
+import (
+	errorsmod "cosmossdk.io/errors"
+
+	channeltypes "github.com/cosmos/ibc-go/v7/modules/core/04-channel/types"
+)
+
+func (suite *TypesTestSuite) TestUpgradeErrorIsOf() {
+	ue := channeltypes.UpgradeError{}
+	suite.Require().True(errorsmod.IsOf(ue, channeltypes.UpgradeError{}))
+	suite.Require().False(errorsmod.IsOf(ue, channeltypes.ErrInvalidChannel))
+
```

Comment:
```suggestion
	ue := channeltypes.NewUpgradeError(1, channeltypes.ErrInvalidChannel)
	suite.Require().True(errorsmod.IsOf(ue, channeltypes.UpgradeError{}))
	suite.Require().False(errorsmod.IsOf(ue, channeltypes.ErrInvalidChannel))
```
what happens in this case? (this is closer to what we would use in practice

Reply:
Ah, this works:
```diff
-       ue := channeltypes.UpgradeError{}
-       suite.Require().True(errorsmod.IsOf(ue, channeltypes.UpgradeError{}))
+       ue := channeltypes.NewUpgradeError(1, channeltypes.ErrInvalidChannel)
+       suite.Require().True(errorsmod.IsOf(ue, channeltypes.NewUpgradeError(1, channeltypes.ErrInvalidChannel)))
        suite.Require().False(errorsmod.IsOf(ue, channeltypes.ErrInvalidChannel))
 
        wrappedErr := errorsmod.Wrap(ue, "wrapped upgrade error")
-       suite.Require().True(errorsmod.IsOf(wrappedErr, channeltypes.UpgradeError{}))
+       suite.Require().True(errorsmod.IsOf(wrappedErr, channeltypes.NewUpgradeError(1, channeltypes.ErrInvalidChannel)))
 }
```

Looks like without a custom `Is` function, it just does a reflect, minus the wrapped string. If the sequences mismatch it'll return false

Reply:
Leaving off the custom `Is` function should work fine